### PR TITLE
[Snyk] Security upgrade transformers from 4.30.2 to 5.0.0rc1

### PR DESCRIPTION
**Pull request by jhamot**
snyk-top-banner

#### Snyk changed the following file(s):

- `requirements.txt`

---

> [!IMPORTANT]
>
> - Check the changes in this PR to ensure they won't cause issues with your project.
> - Max score is 1000. Note that the real score may have changed since the PR was raised.
> - This PR was automatically created by Snyk using the credentials of a real user.
> - Some vulnerabilities couldn't be fully fixed and so Snyk will still find them when the project is tested again. This may be because the vulnerability existed within more than one direct dependency, but not all of the affected dependencies could be upgraded.

---

**Note:** _You are seeing this because you or someone else with access to this repository has authorized Snyk to open fix PRs._

For more information: 
🧐 [View latest project report](https://app.snyk.io/org/lucasfilm/project/097774dc-056b-450a-947b-0ba352e607d9?utm_source&#x3D;github&amp;utm_medium&#x3D;referral&amp;page&#x3D;fix-pr)
📜 [Customise PR templates](https://docs.snyk.io/scan-using-snyk/pull-requests/snyk-fix-pull-or-merge-requests/customize-pr-templates?utm_source=github&utm_content=fix-pr-template)
🛠 [Adjust project settings](https://app.snyk.io/org/lucasfilm/project/097774dc-056b-450a-947b-0ba352e607d9?utm_source&#x3D;github&amp;utm_medium&#x3D;referral&amp;page&#x3D;fix-pr/settings)
📚 [Read about Snyk's upgrade logic](https://docs.snyk.io/scan-with-snyk/snyk-open-source/manage-vulnerabilities/upgrade-package-versions-to-fix-vulnerabilities?utm_source=github&utm_content=fix-pr-template)

---

**Learn how to fix vulnerabilities with free interactive lessons:**

🦉 [Deserialization of Untrusted Data](https://learn.snyk.io/lesson/insecure-deserialization/?loc&#x3D;fix-pr)

[//]: # 'snyk:metadata:{"breakingChangeRiskLevel":null,"FF_showPullRequestBreakingChanges":false,"FF_showPullRequestBreakingChangesWebSearch":false,"customTemplate":{"variablesUsed":[],"fieldsUsed":[]},"dependencies":[{"name":"transformers","from":"4.30.2","to":"5.0.0rc1"}],"env":"prod","issuesToFix":["SNYK-PYTHON-TRANSFORMERS-14564337"],"prId":"77374149-3f5e-40bf-892e-1f9e684de366","prPublicId":"77374149-3f5e-40bf-892e-1f9e684de366","packageManager":"pip","priorityScoreList":[149],"projectPublicId":"097774dc-056b-450a-947b-0ba352e607d9","projectUrl":"https://app.snyk.io/org/lucasfilm/project/097774dc-056b-450a-947b-0ba352e607d9?utm_source=github&utm_medium=referral&page=fix-pr","prType":"fix","templateFieldSources":{"branchName":"default","commitMessage":"default","description":"default","title":"default"},"templateVariants":["updated-fix-title","priorityScore"],"type":"auto","upgrade":[],"vulns":["SNYK-PYTHON-TRANSFORMERS-14564337"],"patch":[],"isBreakingChange":false,"remediationStrategy":"vuln"}'

### CODE REVIEW

**Comment by jhamot** [comment]
Merge Risk: High

This is a major version upgrade to a release candidate that introduces significant and intentional breaking changes. The library is now PyTorch-first, deprecating explicit support for TensorFlow and Flax in many areas. Key changes affect tokenization, internal HTTP clients, and several model-specific features.  
  
**Highlights:**  
- **Tokenizer Unification**: The distinction between "Fast" (Rust-based) and "Slow" (Python-based) tokenizers has been removed. This simplifies the API but may affect custom tokenizer implementations. The `use_fast` flag is obsolete.  
- **HTTP Client Change**: The underlying HTTP client has switched from `requests` to `httpx`. Error handling that catches `requests.HTTPError` must be updated to catch `httpx.HTTPError` or the more compatible `huggingface_hub.HfHubHTTPError`.  
  
Source: Hugging Face GitHub Issues & Migration Guide  
**Recommendation:** Review the official v5 migration guide carefully. Update error handling for HTTP requests and validate all tokenizer and model loading logic before merging.

> **Notice 🤖:** This content was augmented using **artificial intelligence**. AI-generated content **may** contain **errors** and should be **reviewed** for accuracy before use.
